### PR TITLE
Using new `UserStore` interface

### PR DESCRIPTION
**Pull request by nhachicha**
⚠️ **THIS REQUIRES `realm-java 3.1.0` TO BE RELEASED FIRST** 

- [x] Fixing duplicate Key creation
- [x] Using unique key _alias_ (per package name)
- [x] Expose KeyStore helper method (Unlock)
- [x] Add more tests coverage
- [x] Update `build.gradle` once `realm-java 3.1.0` is released

### CODE REVIEW

File: app/src/main/java/io/realm/android/SecureUserStore.java [hunk 1]
```diff
@@ -1,19 +1,3 @@
-/*
```
Comment (Zhuinden) [inline app/src/main/java/io/realm/android/SecureUserStore.java]: Huh? Where'd the license go?

File: app/src/main/java/io/realm/android/SecureUserStore.java [hunk 2]
```diff
@@ -22,151 +6,97 @@
...
-        } else {
-            return null;
+            RealmLog.error(e);
         }
     }
 
     /**
      * Retrieves the {@link SyncUser} by decrypting first the serialised Json.
-     * @param key the {@link SharedPreferences} key.
      * @return the {@link SyncUser} with the given key.
```
Comment (Zhuinden) [inline app/src/main/java/io/realm/android/SecureUserStore.java]: I don't see the key, only current user

File: app/src/main/java/io/realm/android/SecureUserStore.java (hunk 1 above)
Comment (nhachicha) [reply app/src/main/java/io/realm/android/SecureUserStore.java]: Good catch! AS going crazy 👻

File: app/src/main/java/io/realm/android/SecureUserStore.java (hunk 2 above)
Comment (nhachicha) [reply app/src/main/java/io/realm/android/SecureUserStore.java]: I think you reviewed the previous commit `ca72a833f86589b6e5c3765382c9d7def7e86bcd`  for this file 😀

Comment (Zhuinden) [reply app/src/main/java/io/realm/android/SecureUserStore.java]: most likely 😄

**Comment by nhachicha** [comment]
fixes https://github.com/realm/realm-java/issues/3743

File: app/src/androidTest/java/io/realm/android/SecureUserStoreTests.java [hunk 3]
```diff
@@ -0,0 +1,198 @@
+/*
+ * Copyright 2016 Realm Inc.
```
Comment (dalinaum) [inline app/src/androidTest/java/io/realm/android/SecureUserStoreTests.java]: 2017?

**Pull request merged**